File: Client/src/com/badlogic/util/Vector.java
package com.badlogic.util;

public class Vector {
    private double x;
    private double y;
    public static final Vector LEFT = new Vector(-1, 0),
                               RIGHT = new Vector(1, 0),
                               UP = new Vector(0, -1),
                               DOWN = new Vector(0, 1);

    public Vector() {
        this.x = 0;
        this.y = 0;
    }

    public Vector(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() { return x; }

    public double getY() {
        return y;
    }

    public void setX(double x) { this.x = x; }

    public void setY(double y) { this.y = y; }

    public void set(Vector vector) {
        this.x = vector.getX();
        this.y = vector.getY();
    }

    public void set(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public Vector sum(Vector vector) {
        return new Vector(this.x + vector.x, this.y + vector.y);
    }

    public Vector difference(Vector vector) {
        return new Vector(this.x - vector.getX(), this.y - vector.getY());
    }

    public Vector difference(double x, double y) {
        return new Vector(this.x - x, this.y - y);
    }

    public void add(double x, double y) {
        this.x += x;
        this.y += y;
    }

    public void subtract(double x, double y) {
        this.x -= x;
        this.y -= y;
    }

    public void add(Vector vector) {
        this.x += vector.getX();
        this.y += vector.getY();
    }

    public Vector multiply(double scalar) {
        return new Vector(this.x * scalar, this.y * scalar);
    }

    public Vector multiply(Vector vector) {
        return new Vector(this.x * vector.getX(), this.y * vector.getY());
    }

    public double getMagnitude() {
        return Math.sqrt(Math.pow(this.x, 2) + Math.pow(this.y, 2));
    }

    public Vector getNormalized() {
        double magnitude = this.getMagnitude();
        return new Vector(this.x / magnitude, this.y / magnitude);
    }

    @Override
    public String toString() {
        return "(x: " + this.x + ", y: " + this.y + ")";
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;

        if (obj == null || obj.getClass() != this.getClass())
            return false;

        var pos = (Vector)obj;
        return this.x == pos.getX() && this.y == pos.getY();
    }

    public static Vector fromPoint(Point point) {
        return new Vector(point.getX(), point.getY());
    }

    public Point getSerializable() {
        return new Point(this.x, this.y);
    }
}

File: Server/Game/Vector.cs
using System;
using System.Runtime.Serialization;
using System.Text;

namespace Server.Game
{
    [DataContract]
    public class Vector
    {
        [DataMember] public double X { get; set; }
        [DataMember] public double Y { get; set; }

        public Vector() { }

        public Vector(double x, double y)
        {
            X = x;
            Y = y;
        }

        public void Add(Vector vector)
        {
            X += vector.X;
            Y += vector.Y;
        }

        public void Add(double scalar)
        {
            X += scalar;
            Y += scalar;
        }

        public static Vector operator+(Vector left, Vector right) =>
            new Vector(left.X + right.X, left.Y + right.Y);

        public static Vector operator *(Vector vector, double scalar) =>
            new Vector(vector.X * scalar, vector.Y * scalar);

        public Vector Normalized()
        {
            var magnitude = Math.Sqrt(Math.Pow(X, 2) + Math.Pow(Y, 2));
            return new Vector(X / magnitude, Y / magnitude);
        }

        public override string ToString()
        {
            var builder = new StringBuilder();
            builder.AppendFormat("X: {0:0.00}, Y: {0:0.00}", X, Y);
            return builder.ToString();
        }
    }
}